aiobotocore/httpsession.py

```diff
@@ -205,7 +204,18 @@ async def send(self, request):
                 data = _IOBaseWrapper(data)
 
             url = URL(url, encoded=True)
-            response = await self._session.request(
+
+            session = self._sessions.get(proxy_url)
+            if session is None:
+                connector = self._create_connector(proxy_url)
+                self._sessions[proxy_url] = session = aiohttp.ClientSession(
+                    connector=connector,
+                    timeout=self._timeout,
+                    skip_auto_headers={'CONTENT-TYPE'},
+                    auto_decompress=False,
+                )
```

Comment:
```suggestion
            if not (session := self._sessions.get(proxy_url)):
                connector = self._create_connector(proxy_url)
                self._sessions[proxy_url] = session = await self._exit_stack.enter_async_context(aiohttp.ClientSession(
                    connector=connector,
                    timeout=self._timeout,
                    skip_auto_headers={'CONTENT-TYPE'},
                    auto_decompress=False,
                ))
```

Then you don't need to iterate over the sessions when doing cleanup

Reply:
Great idea! I added the `AsyncExitStack` over at #1077 and will push here, too.